src/libraries/System.Text.Json/tests/Common/JsonSchemaExporterTests.TestTypes.cs

```diff
@@ -0,0 +1,1320 @@
+                            GetCustomAttribute<RegularExpressionAttribute>(ctx.PropertyInfo?.AssociatedParameter?.AttributeProvider);
+
+                        if (regexAttr != null)
+                        {
+                            schema["pattern"] = regexAttr.Pattern;
+                        }
+                    }
+                });
+
+            yield return new TestData<ClassWithJsonPointerEscapablePropertyNames>(
+                Value: new ClassWithJsonPointerEscapablePropertyNames { Value = new() },
+                ExpectedJsonSchema: """
+                {
+                    "type": ["object","null"],
+                    "properties": {
+                        "~/path/to/value": {
+                            "type": "object",
+                            "properties": {
+                                "Value" : {"type":"integer"},
+                                "Next": {
+                                    "type": ["object","null"],
+                                    "properties": {
+                                        "Value" : {"type":"integer"},
+                                        "Next": {"$ref":"#/properties/~0~1path~1to~1value/properties/Next"}
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                """);
+
+            // Collection types
+            yield return new TestData<int[]>([1, 2, 3], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"integer"}}""");
+            yield return new TestData<List<bool>>([false, true, false], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"boolean"}}""");
+            yield return new TestData<HashSet<string>>(["one", "two", "three"], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":["string","null"]}}""");
+            yield return new TestData<Queue<double>>(new([1.1, 2.2, 3.3]), ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"number"}}""");
+            yield return new TestData<Stack<char>>(new(['x', '2', '+']), ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"string"}}""");
+            yield return new TestData<ImmutableArray<int>>([1, 2, 3], ExpectedJsonSchema: """{"type":"array","items":{"type":"integer"}}""");
+            yield return new TestData<ImmutableList<string>>(["one", "two", "three"], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":["string","null"]}}""");
+            yield return new TestData<ImmutableQueue<bool>>([false, false, true], ExpectedJsonSchema: """{"type":["array","null"],"items":{"type":"boolean"}}""");
+            yield return new TestData<object[]>([1, "two", 3.14], ExpectedJsonSchema: """{"type":["array","null"],"items":true}""");
+            yield return new TestData<System.Collections.ArrayList>([1, "two", 3.14], ExpectedJsonSchema: """{"type":["array","null"],"items":true}""");
```

Comment:
`items: true` is unnecessary.  It's the same as just omitting `items`.

Reply:
I've already some post-processing transformations that convert `{ }` to `true` and `{ "not" : true }` to `false`, so it might make sense to do that same for `"items" : true` or `anyOf : [.., true, ..]`